### PR TITLE
Add activecontrols plugin - highlights active toolbar buttons

### PR DESCRIPTION
**Pull request by ra100**
Plugin adds class "active" to buttons in toolbar pane based on position of cursor in editor.
Partly solves issue #12

### CODE REVIEW

File: plugins/activecontrols/trumbowyg.activecontrols.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                        if (t.selection !== null) {
+                            t.o.controlButtonActivate(t.selection.commonAncestorContainer.parentNode, t);
+                        }
+                    }
+                }
+            },
+            activeTags: [],
+            // Active tags
+            controlButtonActivate: function (element, t) {
+                var tagToButton = {
```
Comment (Alex-D) [inline plugins/activecontrols/trumbowyg.activecontrols.js]: maybe add that to $.trumbowyg object to made other plugins able to add others tags for other buttons

File: plugins/activecontrols/trumbowyg.activecontrols.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                    }
+                }
+                t.o.activeTags = tags;
+            },
+            getTagsRecursive: function (element, tags) {
+                var tag = element.tagName;
+                if (tag === 'DIV')
+                    return;
+                if (tag === 'P') {
+                    if (element.style.textAlign !== "") {
```
Comment (Alex-D) [inline plugins/activecontrols/trumbowyg.activecontrols.js]: if (tag === 'P' && element.style.textAlign !== '') {

insteed of two imbricated if would be better I think :)

**Comment by Alex-D** [comment]
I test it this asap because it's a must have for a WYSIWYG. Maybe I could increase the Trumbowyg core size to add this to the core...

Thanks again!

File: plugins/activecontrols/trumbowyg.activecontrols.js (hunk 2 above)
Comment (ra100) [inline plugins/activecontrols/trumbowyg.activecontrols.js]: Right, I must have tried something where I needed nested conditions, well, seems like obsolete now.

Comment (Alex-D) [inline plugins/activecontrols/trumbowyg.activecontrols.js]: No problem, I understand :)

**Comment by ra100** [comment]
I'm happy that I can contribute.
And now I realized, that in case there are formatting buttons in dropdown, they are not highlighted. So maybe I should change format of `tagToButton` to use arrays, like this:
```
tagToButton = {
  'strong': ['strong'],
  'b': ['strong'],
  'h1': ['h1', 'formatting'],
```
if it's not overkill

File: plugins/activecontrols/trumbowyg.activecontrols.js (hunk 1 above)
Comment (ra100) [inline plugins/activecontrols/trumbowyg.activecontrols.js]: I'm not sure how should I do it right: `$.trumbowyg.tagToButton =` or `t.o.tagToButton`, or just add tagToButton to trumbowyg.js?

Comment (Alex-D) [inline plugins/activecontrols/trumbowyg.activecontrols.js]: `$.trumbowyg.tagToButton` is better.
Look at upload plugin :)

**Comment by agwidarsito** [comment]
Looks great! Let's get this in! :+1:

**Comment by jakejarrett** [comment]
This would be a great addition! [LGTM](https://media.giphy.com/media/13HBDT4QSTpveU/giphy.gif)

**Comment by ra100** [comment]
I see this is already implemented in core, so if I'm right, whole PR can be closed/removed.

**Comment by Alex-D** [comment]
Yes! I've adapted your into the core and make it extensible for plugins :)

Thanks for your PR which help me a lot!

**Pull request closed without merging**